medium-tests/src/test/java/mediumtest/tracking/IssueTrackingMediumTests.java

```diff
@@ -71,6 +78,155 @@ void stop() {
     }
   }
 
+  @Test
+  void it_should_track_issue_secondary_locations(@TempDir Path baseDir) {
+    var filePath = createFile(baseDir, "Foo.java",
+      "package devoxx;\n" +
+        "\n" +
+        "public class Foo {\n" +
+        "  public void run() {\n" +
+        "    prepare(\"action1\");\n" +
+        "    execute(\"action1\");\n" +
+        "    release(\"action1\");\n" +
+        "  }\n" +
+        "}\n");
+    var projectKey = "projectKey";
+    var connectionId = "connectionId";
+    var branchName = "main";
+    var ruleKey = "java:S1192";
+    var message = "Define a constant instead of duplicating this literal \"action1\" 3 times.";
+
+    var fileUri = filePath.toUri();
+    var client = newFakeClient()
+      .withInitialFs(CONFIG_SCOPE_ID, baseDir, List.of(new ClientFileDto(fileUri, baseDir.relativize(filePath), CONFIG_SCOPE_ID, false, null, filePath, null, null)))
+      .build();
+    var server = newSonarQubeServer("9.5")
+      .withProject(projectKey)
+      .withQualityProfile("qp", qualityProfile -> qualityProfile.withLanguage("java")
+        .withActiveRule(ruleKey, activeRule -> activeRule.withSeverity(IssueSeverity.MAJOR)))
+      .start();
+    backend = newBackend()
+      .withSonarQubeConnection(connectionId, server,
+        storage -> storage.withPlugin(TestPlugin.JAVA).withProject(projectKey,
+          project -> project.withRuleSet("java", ruleSet -> ruleSet.withActiveRule(ruleKey, "MINOR"))
+            .withMainBranch(branchName)))
+      .withStandaloneEmbeddedPluginAndEnabledLanguage(TestPlugin.JAVA)
+      .build(client);
+
+    backend.getConfigurationService()
+      .didAddConfigurationScopes(new DidAddConfigurationScopesParams(List.of(
+        new ConfigurationScopeDto(CONFIG_SCOPE_ID, null, true, CONFIG_SCOPE_ID,
+          new BindingConfigurationDto(connectionId, projectKey, true)))));
+
+    var url = "/batch/issues?key=" + UrlUtils.urlEncode(projectKey + ":" + filePath) + "&branch=" + branchName;
+    var response = ScannerInput.ServerIssue.newBuilder()
+      .setKey("uuid")
+      .setRuleRepository("java")
+      .setRuleKey("S1192")
+      .setChecksum("395d7a96efa8afd1b66ab6b680d0e637")
+      .setMsg(message)
+      .setLine(5)
+      .setCreationDate(123456789L)
+      .setPath("Foo.java")
+      .setType("BUG")
+      .setManualSeverity(true)
+      .setSeverity(Constants.Severity.BLOCKER)
+      .build();
+    server.getMockServer().stubFor(get(url).willReturn(aResponse().withStatus(200).withResponseBody(protobufBodyDelimited(response))));
```

Comment:
This could be moved to the server fixture instead of manually mocking it here. And there is maybe already a method there to configure that